apps/api/src/auth/auth.service.ts

```diff
@@ -0,0 +1,131 @@
+import {Injectable, Logger} from '@nestjs/common';
+import {PrismaService} from '../prisma.service';
+import {Prisma} from '@prisma/client';
+import {randomBytes} from 'crypto';
+import {addMinutes} from 'date-fns';
+import crypto from 'crypto';
+import {JwtService} from '@nestjs/jwt';
+import {AuthMapper} from './auth.mapper';
+import {TokenDto} from './dto/token.dto';
+import {
+  AuthOtpCreatedEvent,
+  AuthSignInEvent,
+  EventPublisher,
+} from '@gddy-coding-exercise/shared-events';
+
+function generateOTP(length: number = 6): string {
+  const pool = '0123456789';
+  const poolLength = pool.length;
+  const bytes = randomBytes(length);
+  return Array.from(bytes)
+    .map((x) => pool[x % poolLength])
+    .join('');
+}
+
+function hashToken(token: string): string {
+  return crypto.createHash('sha256').update(token).digest('hex');
+}
```

Comment:
Hashes tokens before storing them in the database. This is probably not necessary since the tokens are very short-lived, but just wanted to demonstrate that I was thinking about security here. Of course, it's kind of silly because we then turn around and emit an event with the plaintext OTP to stick in a Kafka topic, but my previous suggestion applies here in that I don't think it's a super big deal because of the short-lived nature of the OTP.